staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +192,28 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// EventHandlerCount return the number of actually registered
+	// event handlers.
+	EventHandlerCount() int
+
+	// IsStopped reports whether the informer has already been stopped.
+	// Adding event handlers to already stopped informers is not possible.
+	IsStopped() bool
+
+	// IsStarted reports whether the informer has already been started
+	IsStarted() bool
+}
+
+// ResourceEventHandlerHandle is a handle returned by the
+// registration methods of SharedInformers for a registered
+// ResourceEventHandler. It can be used later on to remove
+// a registration again.
+// This indirection is required, because the ResourceEventHandler
+// interface can be implemented by non-go-comparable handlers, which
+// could not be removed from a list anymore.
+type ResourceEventHandlerHandle struct {
+	ResourceEventHandler
```

Comment:
is there a reason we need to make this public and inline it? making it private and not inlined makes the handle more opaque outside this package, which seems good

Reply:
I did the inlining to avoid changes to the lower coding layers. But you are right, it should not be exposed to the caller, because then he could change it.

There are two possibilities to keep the lower layers:
- make the field private and implement the handler interface for the handle (that's what I've done with the next commit)
- add another invisible indirection layer by internally registering a wrapped handler and return a handle referring to the wrapped handler to be able to remove it again

What do you prefer?

Reply:
yeah, if the handle just contained the listener pointer as a private field, it wouldn't need to implement or expose anything, and could be an opaque pure handle

Reply:
done